### ISSUE TITLE
Glyph placing is odd

### ISSUE THREAD

**Issue by davidgiga1993** [maltaisn/msdf-gdx-gen#5]
I've noticed that using roboto-regular the glyph placing is a bit odd.

This is a sample render using the shader from msdf-gen and the texture has been generated with this utility.

![image](https://user-images.githubusercontent.com/3924958/145730632-894ae295-609a-4195-a4f9-2d0986a9ff0a.png)

As you can see the `B` is placed a bit too high as well as the `l` and the `L`. The `D` also looks a bit off.
The issues persists when scaling the font to any size, so my guess is that the font generation messes something up?

**Comment by maltaisn** [comment]
I can reproduce this issue with roboto-regular with 32 px size. The issue disappears at 48 px size. 

It's been some time since I've taken at look at this, but it seems I knew this was an issue judging by this comment:
https://github.com/maltaisn/msdf-gdx-gen/blob/ae0fed1984188b2816b62a3a58088e0f691c14a7/gen/src/main/kotlin/com/maltaisn/msdfgdx/gen/BMFont.kt#L49-L52

The issue gets worse at smaller font sizes. One solution is to manually adjust the `yoffset` field for problematic characters in the generated .fnt file. For example, the `t` character (id=116) has a very noticeable offset in the roboto-16 example font, but it looks fine after changing `yoffset=4` to `yoffset=3` (smaller number moves the glyph up).

**Comment by davidgiga1993** [comment]
I already tried adjusting the yoffset but 1px step size is not fine granular enough. That's probably why the issue disappears with larger fonts. A proper solution would be to use float as yoffset (and make libgdx use float as well).
Since I'm already running my own fork of libgdx I'll do some experiments, maybe that already solves the issue?

**Comment by rafaskb** [comment]
Hey @davidgiga1993 were you able to fix that issue by any chance? I'm running into the same problem and everything points to msdf-gdx-gen being the culprit (Hiero always outputs a valid font, even with SDF).

I'm trying to understand the code to maybe try some things and hopefully fix it, but I'm having a hard time understanding where this offset is calculated. Would appreciate if anyone could point me to some directions here.

**Comment by davidgiga1993** [comment]
> Hey @davidgiga1993 were you able to fix that issue by any chance? I'm running into the same problem and everything points to msdf-gdx-gen being the culprit (Hiero always outputs a valid font, even with SDF).
> 
> I'm trying to understand the code to maybe try some things and hopefully fix it, but I'm having a hard time understanding where this offset is calculated. Would appreciate if anyone could point me to some directions here.

Nope, not really. I moved away from the idea of using msdf and rather generate a dynamic font texture atlas on-demand which works well enough for my use case
